src/pudl_rmi/coordinate.py

```diff
@@ -0,0 +1,274 @@
+        if not is_path_a_good_file(file_path) or clobber_any:
+            logger.info(
+                "Deprish to FERC1 granular connection not found at "
+                f"{file_path}. Generating a new output."
+            )
+            connects_deprish_ferc1 = pudl_rmi.connect_deprish_to_ferc1.execute(
+                plant_parts_eia=self.grab_plant_part_list(
+                    clobber=clobber_plant_part_list),
+                deprish_eia=self.grab_deprish_to_eia(
+                    clobber=clobber_deprish_eia),
+                ferc1_to_eia=self.grab_ferc1_to_eia(
+                    clobber=clobber_ferc1_eia),
+                clobber=clobber
+            )
+            # export
+            connects_deprish_ferc1.to_pickle(pudl_rmi.FILE_PATH_DEPRISH_FERC1)
+
+        else:
+            logger.info(
+                f"Reading the depreciation to FERC connection from {file_path}"
+            )
+            connects_deprish_ferc1 = pd.read_pickle(file_path)
+        return connects_deprish_ferc1
+
+    def grab_all(self, clobber_all=False):
+        """
+        Gotta catch em all. Get all of the RMI outputs.
+
+        Get or regenerate all of the RMI outputs. This method is mostly for
+        testing purposes because it returns all 5 outputs. To grab individual
+        outputs, it is recommended to use the output-specific method.
+
+        Args:
+            clobber_all (boolean): Deafult is False, which will grab the
+                outputs if they already exist, or generate them if they don't
+                exist. True will re-generate the outputs whether they exist on
+                disk. Re-generating everything will take ~15 minutes.
+
+        Returns:
+            pandas.DataFrame: EIA plant-part list - table of "plant-parts"
+                which are groups of aggregated EIA generators that coorespond
+                to portions of plants from generators to fuel types to whole
+                plants.
+            pandas.DataFrame: a table of depreciation studies. These records
+                have been cleaned and standardized with plant's "common" lines
+                allocated across their cooresponding plant records.
+            pandas.DataFrame: a table of the connection between the
+                depreciation studies and the EIA plant-parts list.
+            pandas.DataFrame: a table of the connection between the FERC1
+                plants and the EIA plant-parts list.
+            pandas.DataFrame: a table of the conneciton between the
+                depreciation studies and the FERC1 plants.
+        """
+        ppl = self.grab_plant_part_list(clobber=clobber_all)
+        d = self.grab_deprish(clobber=clobber_all)
+        de = self.grab_deprish_to_eia(clobber=clobber_all)
+        fe = self.grab_ferc1_to_eia(clobber=clobber_all)
+        df = self.grab_deprish_to_ferc1(clobber=clobber_all)
+        return ppl, d, de, fe, df
+
+
+def is_path_a_good_file(file_path: Path) -> bool:
```

Comment:
I think this implements the logic I was suggesting but I'm finding it pretty confusing.
* What does it mean for the file to be "good"?
* If the file doesn't exist then `is_file()` will return `False` and... that's fine? So the file isn't good but that's not a problem?
* The way it's being used in the functions implies that it's going to return true or false, but it's main utility is to raise an assertion if there's a problem.
* The combination of using this function and `clobber` in a single boolean expression in the functions where it's called is confusing.
* All this in combination makes me unable to read what's happening in the program's flow control.

I think it would be much clearer to separate the two concerns:
1. check whether all is as you expect at the file path (and do nothing if it is)
2. Decide what to do based on whether the path exists and the value of `clobber`

```py
def check_is_file_or_not_exists(path: Path):
    if path.exists() and not path.is_file():
        raise AssertionError(
            f"Path exists but is not a file. Check if {file_path} is a "
            "directory. It should be either a pickled file or nothing."
        )
```
And then in usage:
```py
check_is_file_or_not_exists(file_path)

if not file_path.exists() or clobber:
    # make the new file at file_path

# do other stuff that assumes the file exists.
```

Reply:
yeah this is fair. I debated with myself whether to go with this method. I went for maximum reduction of code but think you are right - it is more clear this way and that far outweighs the reduction of a few lines.